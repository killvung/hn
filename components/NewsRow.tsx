import * as React from 'react';
import { GovUKLink, GovUKNextLink } from './GovUKLink';

interface NewsRowProps {
    title: string
    score: number
    author: string
    timestamp: number
    descendants: number
    url: string
    rootId: string
}

const constructSubHeader = (
    score: number,
    author: string,
    timestamp: number,
    descendants: number,
    rootId: string
) => (
    <div>
        {/* TODO: Construct href and convert timestamp */}
        {score} points by {author} at {timestamp} | <GovUKNextLink href={`/${rootId}`}>{`${descendants} comment`}</GovUKNextLink>
    </div>
)
const NewsRow: React.FC<NewsRowProps> = ({ title, url, score, author, timestamp, descendants, rootId }) => {
    return (
        <div>
            <h2><GovUKLink href={url}>{title}</GovUKLink></h2>
            {constructSubHeader(score, author, timestamp, descendants, rootId)}
        </div>
    )
}

export default NewsRow;
