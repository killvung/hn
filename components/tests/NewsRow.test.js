import React from 'react';
import renderer from 'react-test-renderer';
import NewsRow from '../NewsRow';
import mockDataNews from './mockDataNews.json';

it('renders correctly', () => {
    const { descendants, score, time, title, url, id, by } = mockDataNews;
    const tree = renderer
        .create(<NewsRow
            author={by}
            descendants={descendants}
            score={score}
            timestamp={time}
            title={title}
            url={url}
            rootId={id}
        />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
