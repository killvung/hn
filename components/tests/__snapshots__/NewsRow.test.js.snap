// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsRow - renders correctly 1`] = `
<div>
  <h2>
    <a
      className="govuk-link govuk-link--no-visited-state"
      href="https://github.com/dotnet/maui"
    >
      .NET MAUI: .NET Multi-Platform App UI
    </a>
  </h2>
  <div>
    129
     points by 
    evo_9
     at 
    1618955017
     | 
    <a
      className="govuk-link govuk-link--no-visited-state"
      href="/26882185"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      62 comment
    </a>
  </div>
</div>
`;
